src/modules/recycling-report/dtos/create-recycling-report.dto.ts

```diff
@@ -1,50 +1,23 @@
-import { Type } from 'class-transformer';
-import {
-  IsArray,
-  IsEnum,
-  IsNotEmpty,
-  IsNumber,
-  IsOptional,
-  IsString,
-  ValidateNested,
-} from 'class-validator';
+import { z } from 'zod';
 
 import { ResidueType } from './residue-type.enum';
 
-export class CreateRecyclingReportDto {
-  @IsNotEmpty()
-  @IsNumber()
-  submittedBy: string;
-
-  @IsNotEmpty()
-  @Type(() => Date)
-  reportDate: Date;
-
-  @IsOptional()
-  @IsString()
-  phone?: string;
-
-  @IsNotEmpty()
-  @IsArray()
-  @ValidateNested({ each: true })
-  @Type(() => MaterialDto)
-  materials: MaterialDto[];
-
-  @IsOptional()
-  @IsString()
-  walletAddress?: string;
-
-  @IsNotEmpty()
-  @IsString()
-  evidenceUrl: string;
-}
-
-class MaterialDto {
-  @IsNotEmpty()
-  @IsEnum(ResidueType)
-  materialType: ResidueType;
-
-  @IsNotEmpty()
-  @IsNumber()
-  weightKg: number;
-}
+const MaterialSchema = z.object({
+  materialType: z.nativeEnum(ResidueType),
+  weightKg: z.number().nonnegative(),
+});
+
+export const CreateRecyclingReportSchema = z.object({
+  submittedBy: z.string().min(1),
+  reportDate: z.preprocess((arg) => {
+    if (typeof arg === 'string' || arg instanceof Date) return new Date(arg);
+  }, z.date()),
+  phone: z.string().optional(),
+  materials: z.array(MaterialSchema),
+  walletAddress: z.string().optional(),
+  evidenceUrl: z.string().url(),
+});
+
+export type CreateRecyclingReportDto = z.infer<
+  typeof CreateRecyclingReportSchema
+>;
```

Comment:
Could you add messages to improve these validations?

`materials: z.array(MaterialSchema).min(1, "At least one material must be submitted"),`
` weightKg: z.number().nonnegative().min(0.01, "Weight must be greater than 0"),`
submittedBy: z.string()
    .min(1, "Submitter name cannot be empty")

and we can improve this validations adding regex etc
`walletaddress: z.string().optional().regex(/^0x[a-fA-F0-9]{40}$/, "Invalid EVM wallet address format").or(z.literal('')),`
phone: z.string().optional().regex(/^\+55\d{2}(9\d{8}|\d{8})$/, "Invalid Brazilian phone number format").or(z.literal('')),

r```
eportDate: z.preprocess((arg) => {
    if (typeof arg === 'string' || arg instanceof Date) return new Date(arg);
  }, z.date().max(new Date(), "Report date cannot be in the future")),
```

```
evidenceUrl: z.string()
    .url("Invalid URL format")
    .refine((url) => url.startsWith("https://"), "Evidence URL must start with https"),
```